dojo/tools/ossindex_devaudit/parser.py

```diff
@@ -0,0 +1,100 @@
+import json
+
+from dojo.models import Finding
+
+
+class OssIndexDevauditParser(object):
+    def __init__(self, json_file, test):
+
+        tree = self.parse_json(json_file)
+
+        if tree:
+            self.items = [data for data in self.get_items(tree, test)]
+        else:
+            self.items = []
+
+    def parse_json(self, json_file):
+        if json_file is None:
+            self.items = []
+            return
+        try:
+            tree = json.load(json_file)
+        except:
+            raise Exception("Invalid format")
+
+        return tree
+
+    def get_items(self, tree, test):
+
+        items = {}
+
+        results = {key: value for (key, value) in tree.items()}
+        for package in results['Packages']:
+            package_data = package['Package']
+            if len(package['Vulnerabilities']) > 0:
+                for vulnerability in package['Vulnerabilities']:
+                    item = get_item(
+                        dependency_name=package_data['name'],
+                        dependency_version=package_data['version'],
+                        dependency_source=package_data['pm'],
+                        vulnerability=vulnerability,
+                        test=test
+                    )
+                    unique_key = vulnerability['id']
+                    items[unique_key] = item
+
+        return items.values()
+
+
+def get_item(dependency_name, dependency_version, dependency_source, vulnerability, test):
+
+    cwe_data = vulnerability.get('cwe', '')
+
+    if cwe_data == '' or cwe_data is None:
+        cwe_text = ''
+        cwe = 1035
+    else:
+        cwe_text = cwe_data
+        cwe = cwe_data.replace('CWE-', '')
+
+    reference = vulnerability.get('reference', '')
+    references = 'https://snyk.io/vuln/search?q={}&type={}'.format(dependency_name, dependency_source) + ' ' + reference

```

Comment:
some findings from the files have already a reference to Snyk DB (see `422fd048-4155-4213-8126-75b4c3135820` for ex). I don't think adding an artificial link to Snyk DB is a good idea. We should keep the data as it is exactly in the files.
This will prevent some pb in the futur like for ex if Snyk change how the URL are build.

Reply:
Snyk link removed